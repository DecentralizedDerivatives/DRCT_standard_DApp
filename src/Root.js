import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
