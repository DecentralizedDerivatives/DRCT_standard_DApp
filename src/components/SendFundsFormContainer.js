import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { reduxForm } from 'redux-form';
import SendFundsFormComponent from './SendFundsFormComponent';

const validate = values => {
  const errors = {};
  if (!values.address) {
    errors.address = 'Required';
  }

  if (!values.amount) {
    errors.amount = 'Required';
  }

  if (!values.currency) {
    errors.currency = 'Required';
  }

  return errors;
};

export let SendFundsFormContainer = props => {
  const submitForm = formValues => {
    console.log('submitting Form: ', formValues);
    props.sendFunds(formValues, props.userAccount);
  };

  return (
    <SendFundsFormComponent
      onSubmit={submitForm}
      handleSubmit={props.handleSubmit}
    />
  );
};

SendFundsFormContainer.propTypes = {
  sendFunds: PropTypes.func.isRequired,
  userAccount: PropTypes.string,
};

const mapStateToProps = state => ({
  userAccount: state.user.userAccount
});

const formConfiguration = {
  form: 'create-contract',
  validate
};

SendFundsFormContainer = reduxForm(formConfiguration)(
  SendFundsFormContainer
);

export default (SendFundsFormContainer = connect(
  mapStateToProps, null)(SendFundsFormContainer));
