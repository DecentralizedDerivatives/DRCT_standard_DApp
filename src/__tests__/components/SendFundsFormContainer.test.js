import React from 'react';
import { shallow } from 'enzyme';
import { initStore } from '../../Root';
import SendFundsFormContainer from '../../components/SendFundsFormContainer';

describe('<SendFundsFormContainer />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const wrapper = shallow(
        <SendFundsFormContainer store={initStore()} />
      ).dive().dive().dive();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
