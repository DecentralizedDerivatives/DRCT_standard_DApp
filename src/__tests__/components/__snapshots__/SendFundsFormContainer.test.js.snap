// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SendFundsFormContainer /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(SendFundsFormContainer)
    anyTouched={false}
    array={
      Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      }
    }
    arrayInsert={[Function]}
    arrayMove={[Function]}
    arrayPop={[Function]}
    arrayPush={[Function]}
    arrayRemove={[Function]}
    arrayRemoveAll={[Function]}
    arrayShift={[Function]}
    arraySplice={[Function]}
    arraySwap={[Function]}
    arrayUnshift={[Function]}
    asyncValidating={false}
    autofill={[Function]}
    blur={[Function]}
    change={[Function]}
    clearAsyncError={[Function]}
    clearFields={[Function]}
    clearSubmit={[Function]}
    clearSubmitErrors={[Function]}
    destroy={[Function]}
    destroyOnUnmount={true}
    dirty={false}
    dispatch={[Function]}
    enableReinitialize={false}
    focus={[Function]}
    forceUnregisterOnUnmount={false}
    form="create-contract"
    getFormState={[Function]}
    initialize={[Function]}
    initialized={false}
    invalid={false}
    keepDirtyOnReinitialize={false}
    persistentSubmitErrors={false}
    pristine={true}
    pure={true}
    registerField={[Function]}
    reset={[Function]}
    resetSection={[Function]}
    setSubmitFailed={[Function]}
    setSubmitSucceeded={[Function]}
    shouldAsyncValidate={[Function]}
    shouldError={[Function]}
    shouldValidate={[Function]}
    shouldWarn={[Function]}
    startAsyncValidation={[Function]}
    startSubmit={[Function]}
    stopAsyncValidation={[Function]}
    stopSubmit={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    submit={[Function]}
    submitFailed={false}
    submitSucceeded={false}
    submitting={false}
    syncErrors={Object {}}
    syncWarnings={Object {}}
    touch={[Function]}
    touchOnBlur={true}
    touchOnChange={false}
    unregisterField={[Function]}
    untouch={[Function]}
    updateSyncErrors={[Function]}
    updateSyncWarnings={[Function]}
    updateUnregisteredFields={false}
    userAccount=""
    valid={true}
    validExceptSubmit={true}
    validate={[Function]}
    values={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "anyTouched": false,
      "array": Object {
        "insert": [Function],
        "move": [Function],
        "pop": [Function],
        "push": [Function],
        "remove": [Function],
        "removeAll": [Function],
        "shift": [Function],
        "splice": [Function],
        "swap": [Function],
        "unshift": [Function],
      },
      "asyncValidate": [Function],
      "asyncValidating": false,
      "autofill": [Function],
      "blur": [Function],
      "change": [Function],
      "clearAsyncError": [Function],
      "clearFields": [Function],
      "clearSubmit": [Function],
      "clearSubmitErrors": [Function],
      "destroy": [Function],
      "dirty": false,
      "dispatch": [Function],
      "error": undefined,
      "form": "create-contract",
      "handleSubmit": [Function],
      "initialValues": undefined,
      "initialize": [Function],
      "initialized": false,
      "invalid": false,
      "pristine": true,
      "pure": true,
      "reset": [Function],
      "resetSection": [Function],
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      },
      "submit": [Function],
      "submitFailed": false,
      "submitSucceeded": false,
      "submitting": false,
      "touch": [Function],
      "triggerSubmit": undefined,
      "untouch": [Function],
      "userAccount": "",
      "valid": true,
      "validate": [Function],
      "warning": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "anyTouched": false,
        "array": Object {
          "insert": [Function],
          "move": [Function],
          "pop": [Function],
          "push": [Function],
          "remove": [Function],
          "removeAll": [Function],
          "shift": [Function],
          "splice": [Function],
          "swap": [Function],
          "unshift": [Function],
        },
        "asyncValidate": [Function],
        "asyncValidating": false,
        "autofill": [Function],
        "blur": [Function],
        "change": [Function],
        "clearAsyncError": [Function],
        "clearFields": [Function],
        "clearSubmit": [Function],
        "clearSubmitErrors": [Function],
        "destroy": [Function],
        "dirty": false,
        "dispatch": [Function],
        "error": undefined,
        "form": "create-contract",
        "handleSubmit": [Function],
        "initialValues": undefined,
        "initialize": [Function],
        "initialized": false,
        "invalid": false,
        "pristine": true,
        "pure": true,
        "reset": [Function],
        "resetSection": [Function],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            },
            "unsubscribe": [Function],
          },
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "unsubscribe": [Function],
        },
        "submit": [Function],
        "submitFailed": false,
        "submitSucceeded": false,
        "submitting": false,
        "touch": [Function],
        "triggerSubmit": undefined,
        "untouch": [Function],
        "userAccount": "",
        "valid": true,
        "validate": [Function],
        "warning": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
